snapshot-labs/pineapple
# fix: catch error when all promises reject

## 🧿 Current issues / What's wrong ?

`Promises.any` is not catch, and will throw an `AggregateError` when all promises reject

## 💊 Fixes / Solution

Handle the `AggregateError`

Fixes #148 
Fixes PINEAPPLE-4

## 🚧 Changes

- Catch the `AggregateError` and return a 400 error code. Will assume that if all gateways fail, the input is probably not valid
- __Add a timeout when requesting gateways__: This will resolve the high memory consumption
- try/catch the AWS set logic
- Use `node-fetch` directly instead of `cross-fetch`
- Remove `bluebird`, and use directly built-in Promise from js

## 🛠️ Tests

- Go to `http://localhost:3000/ipfs/QmS8JB9s69bWLR7GQHQU6D15jV3LuvAuSStC6b1UwyYSk8`
- It should print something
- Go to `http://localhost:3000/ipfs/test`
- It should return a 400 error
